TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs

```diff
@@ -833,5 +835,46 @@ public static void CheckLibrary([NotNull] ProcedureDivision procedureDivision)
 	}
 }
 
-
+    public class SetStatementChecker
+    {
+        public static void OnNode(Node node)
+        {
+            var statement = node.CodeElement as SetStatementForIndexes;
+            if (statement != null)
+            {
+                // Check receivers (incremented)
+                var receivers = node["receivers"] as List<DataDefinition>;
+                if (receivers != null && receivers.Any(x => x.Usage == DataUsage.Pointer))
```

Comment:
use the new StorageAreaWritesDataDefinition directly, you will avoid the use of a generic property `receivers`  that is made for code generation.